plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/utils/get-product-types/index.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+
+type ProductTypeProps = {
+	value: string;
+	label: string;
+};
+
+const productTypes = getSetting< Record< string, string > >(
+	'productTypes',
+	{}
+);
+
+/**
+ * Build product types collection for product types.
+ *
+ * @return {ProductTypeProps[]} Product types collection.
+ */
+export default function getProductTypeOptions(): ProductTypeProps[] {
+	return Object.keys( productTypes ).map( ( key ) => ( {
```

Comment:
I think in some cases it would be better if this was stored as an object instead of an array. For example, this would allow us to do `productTypes[ activeProductType ]` to get the label, instead of having to search through the array. What do you think?

Reply:
Sometimes it's better to focus on the collection; sometimes on the object. 🙂 For instance, the collection fits better when rendering the toolbar button.

Reply:
Makes sense, let's leave it as-is.